cms/djangoapps/contentstore/proctoring.py

```diff
@@ -0,0 +1,106 @@
+"""
+Code related to the handling of Proctored Exams in Studio
+"""
+
+import logging
+
+from django.conf import settings
+
+from xmodule.modulestore.django import modulestore
+
+from edx_proctoring.api import (
+    get_exam_by_content_id,
+    update_exam,
+    create_exam,
+    get_all_exams_for_course,
+)
+from edx_proctoring.exceptions import (
+    ProctoredExamNotFoundException
+)
+
+log = logging.getLogger(__name__)
+
+
+def register_proctored_exams(course_key):
+    """
+    This is typically called on a course published signal. The course is examined for sequences
+    that are marked as timed exams. Then these are registered with the edx-proctoring
+    subsystem. Likewise, if formerly registered exams are unmarked, then those
+    registred exams are marked as inactive
+    """
+
+    if not settings.FEATURES.get('ENABLE_PROCTORED_EXAMS'):
+        # if feature is not enabled then do a quick exit
+        return
+
+    course = modulestore().get_course(course_key)
+    if not course.enable_proctored_exams:
+        # likewise if course does not have this feature turned on
+        return
+
+    # get all sequences, since they can be marked as timed/proctored exams
+    timed_exams = modulestore().get_items(
```

Comment:
AFAIK, Studio doesn't have draft versions of sequences, just verticals and components.

Good question on #2, I'll write a unit test to confirm behavior.